Frameworks/CoreText/CTFontDescriptor.mm

```diff
@@ -111,21 +176,28 @@ CTFontDescriptorRef CTFontDescriptorCreateMatchingFontDescriptor(CTFontDescripto
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CFDictionaryRef CTFontDescriptorCopyAttributes(CTFontDescriptorRef descriptor) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    CF_OBJC_FUNCDISPATCHV(CTFontDescriptorGetTypeID(), CFDictionaryRef, (UIFontDescriptor*)descriptor, fontAttributes);
+    __CFGenericValidateType(descriptor, CTFontDescriptorGetTypeID());
+    return CFDictionaryCreateCopy(CFGetAllocator(descriptor->_dictionary), descriptor->_dictionary);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CFTypeRef CTFontDescriptorCopyAttribute(CTFontDescriptorRef descriptor, CFStringRef attribute) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    CF_OBJC_FUNCDISPATCHV(CTFontDescriptorGetTypeID(), CFTypeRef, (UIFontDescriptor*)descriptor, objectForKey
+                          : static_cast<NSString*>(attribute));
+    __CFGenericValidateType(descriptor, CTFontDescriptorGetTypeID());
+
+    // Use CFRetain instead of a true copy here
+    //  - A generic 'CFCopy' is not available, as not all types are copyable
```

Comment:
>CFCopy [](start = 21, length = 6)

not sure if thats true. I seem to remember a copy being there but in a confusing way / name. But if this matches the ref plat. #yolo?